crates/core/plugin_sm/src/plugin.rs

```diff
@@ -466,3 +463,21 @@ impl Plugin for ExternalPluginCommand {
         }
     }
 }
+
+pub fn deserialize_module_info(
+    module_type: String,
+    records: &mut Reader<&[u8]>,
+) -> Result<Vec<SoftwareModule>, SoftwareError> {
```

Comment:
I would change the signature of this function to take an unparsed stream of bytes rather than CSV records.

```suggestion
pub fn deserialize_module_info(
    module_type: String,
    records: impl std::io::Read,
) -> Result<Vec<SoftwareModule>, SoftwareError> {
```

This change:
* makes the function better encapsulating the format actually used
* removes the need for a `&mut`
* simplifies the tests.